code/modules/power/tesla/coil.dm

```diff
@@ -66,6 +75,19 @@
 	else
 		..()
 
+/obj/machinery/power/tesla_coil/proc/zap()
+	if((last_zap + zap_cooldown) > world.time)
+		return FALSE
+	last_zap = world.time
+	var/coeff = (12 - (input_power_multiplier * 3))
+	coeff = max(coeff, 1)
+	var/shock_coeff = (5 - input_power_multiplier)
+	shock_coeff = max(shock_coeff, 1)
+	var/power = (powernet.avail/coeff)
+	add_load(power)
+	playsound(src.loc, 'sound/magic/LightningShock.ogg', 100, 1, extrarange = 5)
+	tesla_zap(src, 10, power/shock_coeff)
```

Comment:
HMMMMMMMMMMMMMMMMMMMMMMMMMMM
How do I fix this.. Eh, I'l check my math later, see if I can balance it out.